lib/external_service_new/shopify_downloader.rb

```diff
@@ -28,37 +29,122 @@ def each_email(id:, &block)
   end
 
   class ShopifyInternalDownloader
+    AFTER_REMOTE_FILTERS = [
+      "average_basket_revenue_min",
+      "average_basket_revenue_max",
+      "created_at_min",
+      "created_at_max",
+      "total_spent_min",
+      "total_spent_max",
+      "orders_count_min",
+      "orders_count_max",
+      "updated_at_min",
+      "updated_at_max"]
+
+    REMOTE_FILTERS = [
+      "city",
+      "country",
+      "province",
+      "province_code",
+      "zip"]
+
+    attr_reader :dispatcher, :logger
     def initialize(dispatcher:, filters:, logger:nil)
       raise "dispatcher must not be nil, found #{dispatcher.inspect}" unless dispatcher
 
-      @dispatcher = dispatcher
-      @filters    = filters
-      @logger     = logger || respond_to?(:logger) ? logger : nil
+      @dispatcher           = dispatcher
+      @remote_filters       = filters.select{|filter| REMOTE_FILTERS.include?(filter.fetch(:code))}
```

Comment:
@filters = filters, there is no need to declare 2 instance variables, we can use a helper method instead (see suggestion below).